### PR TITLE
Artificial Neural Networks

### PR DESCRIPTION
**Pull request by KilnOfTheSecondFlame**

### CODE REVIEW

File: FS19/ML/Zusammenfassung_ML.tex [hunk 1]
```diff
@@ -2811,5 +2811,101 @@ \subsection{How to Approach a Machine Learning Project}
     \item Learn and improve
 \end{enumerate}
 
+\section{Artificial Neural Networks}
+
+Artificial neural networks try to imitate a biological neural system. The smallest subunit of it is called a neuron, which combines one or multiple input signals from preceding neurons and based on these signals computes an output signal which is propagated to selected neighbouring neurons. A large number of neurons in multiple layers is able to solve complex problems.
```
Comment (gathurian) [inline FS19/ML/Zusammenfassung_ML.tex]: Mah boy why you gotta do so goddamn long sentences??

The smallest subunit of it is called a neuron. _A neuron_ combines one or multiple input [...]

File: FS19/ML/Zusammenfassung_ML.tex [hunk 2]
```diff
@@ -2811,5 +2811,101 @@ \subsection{How to Approach a Machine Learning Project}
...
+\subsection{The Artificial Neuron}
+
+\begin{figure}[tbh!]
+    \centering
+    \includegraphics[width=0.6\linewidth, keepaspectratio]{Pictures/artificial_neuron_layers}
+    \caption{Slice of an input and output layer in an artificial neurol network}
+    \label{fig:artificialneuronlayers}
+\end{figure}
+
+Each neuron fires only if the calculated signal from all the inputs is over a threshold $\theta$. For each neuron $j$ the weighted sum of the $n$ input signal $\text{net}_j$ is calculated as
```
Comment (gathurian) [inline FS19/ML/Zusammenfassung_ML.tex]: For each neuron $j$ **,** the weighted sum [...]

File: FS19/ML/Zusammenfassung_ML.tex [hunk 3]
```diff
@@ -2908,4 +2925,212 @@ \subsection{Description of a Neural Network using Matrices}
...
+Input nodes do no computation, they are just used as input. All output nodes compute their output using the same activation function $\sigma$ (either the Heaviside Step function or the Sigmoid function). These networks are used as binary classifiers if the problem is linearly separable and called Perceptrons.
+
+\subsubsection{Learning Rules - Weights Update}
+
+The update of the weight $w_{ji}$ from input $x_i$ to output $y_j$ with the learning rate $\eta$ being a constant typically between $0 ... 0.5$
+\begin{equation}
+    \Delta w_{ji} = \eta\cdot y_j \cdot x_i
+\end{equation}
+
+When we factor in the error $e_j$ to account for a target value $t_j$ too high or too low than the actual output $y_j$, we get what is called the \textbf{$\delta$-rule} by Widrow and Hoff or \textbf{Hebb's learning rule} by Bürgler.
```
Comment (gathurian) [inline FS19/ML/Zusammenfassung_ML.tex]: _higher or lower than the actual input_ or _too high or too low for the actual input_?

File: FS19/ML/Zusammenfassung_ML.tex [hunk 4]
```diff
@@ -2908,4 +2925,212 @@ \subsection{Description of a Neural Network using Matrices}
...
+e_1 = t - y_1 &= 1 - 1 = 0\\
+\eta \cdot e_1 = \eta (t - y_1) &= 0.3\cdot 0 = 0\\
+w'_{10} = w_{10} + \Delta w_{10} &= -1 + 0.3\cdot 0 \cdot 1 = -1 + 0 = -1\\
+w'_{11} = w_{11} + \Delta w_{11} &= 1 + 0.3\cdot 0 \cdot 1 = 1 + 0 = 1 \\
+w'_{12} = w_{12} + \Delta w_{12} &= 0.3 + 0.3\cdot 0 \cdot 1 = 0.3 + 0 = 0.3 \\
+\end{align*}
+
+\subsubsection{Updating Weights Using Vectors}
+
+Given the input vector $\vec{x}$ and the weight vector $w_j$ for a given node $j$ we can write the output after the activation function $h$ with $n$ input nodes as follows
```
Comment (gathurian) [inline FS19/ML/Zusammenfassung_ML.tex]: [...] for a given node $j$ **,** we can write [...]

File: FS19/ML/Zusammenfassung_ML.tex [hunk 5]
```diff
@@ -2908,4 +2925,212 @@ \subsection{Description of a Neural Network using Matrices}
...
+\end{equation}
+
+\noindent
+Therefore we can write the output as
+\begin{equation}
+y_j = h(\vec{w}_j \cdot \vec{x}) = \left\{ \begin{matrix} 1 & \vec{w}_j \cdot \vec{x} \geq 0\\ 0 & \vec{w}_j \cdot \vec{x} < 0 \end{matrix} \right.
+\end{equation}
+
+\noindent
+And the update of the weights $\vec{w}'_j$ written as follows
```
Comment (gathurian) [inline FS19/ML/Zusammenfassung_ML.tex]: [...] weights $\vec{w}'_j$ _is_ / _can be_ written [...]

**Review by gathurian — CHANGES_REQUESTED** [review]

File: FS19/ML/Zusammenfassung_ML.tex (hunk 1 above)
Comment (KilnOfTheSecondFlame) [reply FS19/ML/Zusammenfassung_ML.tex]: Because we are not in Kindergarten anymore, no need for simple sentences for Babby Trump

Comment (gathurian) [reply FS19/ML/Zusammenfassung_ML.tex]: I don't even understand the damn subject, so let me have it explained like I'm a babby.

**Review by gathurian — APPROVED** [review]

**Pull request merged**